tests/test_driver_alchemy_crud.py

```diff
@@ -1,725 +1,708 @@
-            SELECT hash_type, hash_value FROM index_record_hash
-        ''')}
 
-        assert did == new_did, 'record id does not match'
-        assert rev != new_rev, 'record revision matches prior'
-        assert update_urls == new_urls, 'record urls mismatch'
-        assert file_name == new_file_name, 'file_name does not match'
-        assert version == new_version, 'version does not match'
+def test_driver_update_record_simple_data(index_driver, database_conn):
+    did, baseid, rev, form = insert_base_data(database_conn)
 
-@util.removes('index.sq3')
-def test_driver_update_fails_with_no_records():
-    '''
+    update_size = 256
+    file_name = 'test'
+    version = 'ver_123'
+    changing_fields = {
+        'file_name': file_name,
+        'version': version,
+        'size': update_size,
+    }
+    index_driver.update(did, rev, changing_fields)
+
+    new_did, new_rev, new_file_name, new_size, new_version = database_conn.execute("""
+        SELECT did, rev, file_name, size, version FROM index_record
+    """).fetchone()
+
+    assert did == new_did, 'record id does not match'
+    assert rev != new_rev, 'record revision matches prior'
+    assert file_name == new_file_name, 'file_name does not match'
+    assert update_size == new_size, 'size does not match'
+    assert version == new_version, 'version does not match'
+
+
+def test_driver_update_record_hashes(index_driver, database_conn):
+    did, _, rev, _ = insert_base_data(database_conn)
+    update_hashes = {
+        'a': '1',
+        'b': '2',
+        'c': '3',
+    }
+    changing_fields = {
+        'hashes': update_hashes,
+    }
+    index_driver.update(did, rev, changing_fields)
+
+    new_hashes = {h: v for h, v in database_conn.execute("""
+        SELECT hash_type, hash_value FROM index_record_hash
+    """)}
+    assert update_hashes == new_hashes, 'hashes do not match'
+
+
+def test_driver_update_record_metadata(index_driver, database_conn):
+    did, _, rev, _ = insert_base_data(database_conn)
+    update_metadata = {
+        'a': 'A',
+        'b': 'B',
+    }
+    changing_fields = {
+        'metadata': update_metadata,
+    }
+    index_driver.update(did, rev, changing_fields)
+
+    new_metadata = database_conn.execute("""
+        SELECT index_metadata FROM index_record
+    """).fetchone()[0]
+
+    assert update_metadata == new_metadata, 'metadata does not match'
+
+
+def test_driver_update_record_release_number_separate(index_driver, database_conn):
+    did, _, rev, _ = insert_base_data(database_conn)
+    update_release_number = '1.0'
+    changing_fields = {
+        'release_number': update_release_number,
+    }
+    index_driver.update(did, rev, changing_fields)
+
+    new_release_number = database_conn.execute("""
+        SELECT release_number FROM index_record
+    """).fetchone()[0]
+
+    count = database_conn.execute("""
+        SELECT count(did)
+        FROM index_record
+        WHERE index_metadata ->> 'test' = 'value'
```

Comment:
This should be checking if the `release_number` key is set, right?

Same question for `test_driver_update_record_release_number_metadata` below this.